detectron2/structures/masks.py

```diff
@@ -199,9 +199,20 @@ def crop_and_resize(self, boxes: torch.Tensor, mask_size: int) -> torch.Tensor:
         output = output >= 0.5
         return output
 
-    def get_bounding_boxes(self) -> None:
-        # not needed now
-        raise NotImplementedError
+    def get_bounding_boxes(self) -> Boxes:
+        """
+        Returns:
+            Boxes: tight bounding boxes around bit masks.
+        """
+        boxes = torch.zeros(self.tensor.shape[0], 4, dtype=torch.float32)
+        x_any = torch.any(self.tensor, dim=1)
+        y_any = torch.any(self.tensor, dim=2)
+        for idx in range(self.tensor.shape[0]):
+            x = torch.where(x_any[idx, :])[0]
+            y = torch.where(y_any[idx, :])[0]
+            if len(x) > 0 and len(y) > 0:
+                boxes[idx, :] = torch.as_tensor([x[0], y[0], x[-1], y[-1]], dtype=torch.float32)
```

Comment:
in detectron2, bounding box coordinates are defined to be floating point numbers that range from 0 to H or W, not H-1 or W-1. See https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/NOTES.md#model for an illustration of coordinates.
So here it should be `x[-1] + 1` and `y[-1] + 1`. Consequently a mask with only one pixel will become a box of size 1x1, which matches our definition.